### PR TITLE
Document catching trace for successful assertions

### LINKED ISSUE #846: Traces of successful runs
Traces should be available for successful runs as well to understand why assertions are not failing.

### PR DESCRIPTION
**Pull request by krajek**
I think this documentation extension closes #846.

### CODE REVIEW

File: docs/_pages/documentation.md [hunk 1]
```diff
@@ -980,6 +980,18 @@ With trace:
...
+
+  ```csharp
+object object1 = [...];
+object object2 = [...];
+var traceWriter = new StringBuilderTraceWriter();            
+object1.Should().BeEquivalentTo(object2, opts => opts.WithTracing(traceWriter));             
+string trace = traceWriter.ToString();
+  ```
+
+Alternatively, you could write your own implementation of ITraceWriter for special purposes e.g. writing to a file.
```
Comment (dennisdoomen) [inline docs/_pages/documentation.md]: 🙃 I would use back-tickets for the reference to `ITraceWriter` so it looks like code.

File: docs/_pages/documentation.md [hunk 2]
```diff
@@ -980,6 +980,18 @@ With trace:
   }
   ```
 
+  By default, trace is displayed only when assertion fails as part of the message. If the trace is needed for successful assertion you can catch it like this:
```
Comment (dennisdoomen) [inline docs/_pages/documentation.md]: 🙃 It's _the trace_
🙃 _successful assertions_

Comment (krajek) [reply docs/_pages/documentation.md]: Thanks, I passed the test through Grammarly. Applied your first comment, but not the other.

File: docs/_pages/documentation.md (hunk 1 above)
Comment (krajek) [reply docs/_pages/documentation.md]: Good catch, thanks

**Comment by krajek** [comment]
I have just fixed styling issues. PR is ready to merge.

**Pull request merged**